File: Year-2022/LeetCode-August-Challenge/KthSmallestElementinaSortedMatrix.java
import java.util.*;
class Solution {
    public int kthSmallest(int[][] matrix, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int row = 0; row < matrix.length; ++row){
            for(int col = 0; col < matrix[0].length; ++col){
                pq.add(matrix[row][col]);
                if(pq.size() > k){
                    pq.remove();
                }
            }
        }
        return pq.peek();
    }
}
public class KthSmallestElementinaSortedMatrix {
    public static void main(String[] args) {
        int[][] matrix = {{1,5,9},{10,11,13},{12,13,15}};
        int k = 8;
        Solution s = new Solution();
        System.out.println(s.kthSmallest(matrix, k));
    }
}

File: Year-2022/LeetCode-August-Challenge/KthSmallestElementinaSortedMatrix.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int kthSmallest(vector<vector<int>>& matrix, int k) {
        priority_queue<int> pq;
        for(int row = 0; row < matrix.size(); ++row){
            for(int col = 0; col < matrix[0].size(); ++col){
                pq.push(matrix[row][col]);
                if(pq.size() > k){
                    pq.pop();
                }
            }
        }
        return pq.top();
    }
};
int main(){
    vector<vector<int>> matrix = {{1,5,9},{10,11,13},{12,13,15}};
    int k = 8;
    Solution s;
    cout<<s.kthSmallest(matrix, k)<<endl;
    return 0;
}